x-pack/plugins/dataset_quality/server/routes/data_streams/get_data_stream_details/index.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient } from '@kbn/core/server';
+import { DataStreamTypes } from '../../../types/default_api_types';
+import { DataStreamDetails } from '../../../../common/api_types';
+import { dataStreamService } from '../../../services';
+
+export async function getDataStreamDetails(args: {
+  esClient: ElasticsearchClient;
+  type: DataStreamTypes;
+  datasetQuery: string;
+}): Promise<DataStreamDetails> {
+  const { esClient, type, datasetQuery } = args;
+
+  if (!datasetQuery) {
+    throw new Error(`Dataset query cannot be empty. Received value '${datasetQuery}'`);
+  }
+
+  const indexSettings = await dataStreamService.getDataSteamIndexSettings(esClient, {
+    type,
+    dataset: `*${datasetQuery}*`,
+  });
+
+  const indexesList = Object.values(indexSettings);
+  if (indexesList.length < 1) {
+    throw new Error('index_not_found_exception');
+  }
+
+  const indexCreationDate = indexesList
+    .map((index) => Number(index.settings?.index?.creation_date))
+    .sort((a, b) => a - b)[0];
+
+  return {
+    createdOn: indexCreationDate,
```

Comment:
Maybe it's just me but this looks really weird
`.indices.getSettings()` will return a record and we are just mapping that to a list of creation dates?  and then sorting that and getting the first one. What if `getSettings` returns more than one index? is that a possible scenario? for example, if we query for `my-index` could we get a response with multiple indices like: `my-index`, `my-index-is-ugly`, `my-index-is-beatiful`, etc being all valid indices for the query `my-index`? Maybe that's a case we should check. Instead of looping the whole Record could we just try to access the dedicated index that match with the `datasetQuery`?

Reply:
The `getSettings` always returns an object of indices. For a Data Stream with more than on backing index, it returns multiple indexes, with settings information for each of them. So we've to find the earliest created backing index and use that as creation date.

<img width="1191" alt="Screenshot 2024-01-05 at 17 02 09" src="https://github.com/elastic/kibana/assets/2748376/7eb30c74-2226-45bc-836c-f6446de1db03">

I am open for ideas if there's a better way to get it.

As for `my-index`, `my-index-is-ugly` and `my-index-is-beatiful` scenario above, we'll always send the complete name of the index to this endpoint. The above will be sent as `logs-my-index-default`, `logs-my-index-is-ugly-default` and `logs-my-index-is-beatiful-default`. I've however removed the wild card wrapping while querying for this endpoint e.g. instead of querying like `logs-*my-index-is-beatiful-default*` it'll query `logs-my-index-is-beatiful-default` to make it stricter, assuming that the frontend will send a complete data stream name to this endpoint.

Reply:
Thanks for the detailed explanation, I'd say let's go for the assumption that the consumer will send the complete data stream name 🚀